File: app/src/main/java/com/mmjang/ankihelper/data/model/UserTagEntity.java
package com.mmjang.ankihelper.data.model;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

/**
 * Room entity for UserTag
 * Replaces UserTag (LitePal) model
 *
 * Table: usertag
 * Primary key: tag (unique tag identifier)
 */
@Entity(tableName = "usertag")
public class UserTagEntity {

    @PrimaryKey
    @NonNull
    @ColumnInfo(name = "tag")
    private String tag = "";

    // Default constructor required by Room
    public UserTagEntity() {
    }

    // Constructor for creating with tag
    @Ignore
    public UserTagEntity(@NonNull String tag) {
        this.tag = tag;
    }

    // Getters and Setters
    @NonNull
    public String getTag() {
        return tag;
    }

    public void setTag(@NonNull String tag) {
        this.tag = tag;
    }
}

File: app/src/main/java/com/lmyby/ankihelper/data/model/UserTagEntity.kt
package com.lmyby.ankihelper.data.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey

/**
 * Room entity for UserTag
 * Replaces UserTag (LitePal) model
 *
 * Table: usertag
 * Primary key: tag (unique tag identifier)
 * Converted to Kotlin as part of Phase 2 data model migration
 */
@Entity(tableName = "usertag")
data class UserTagEntity(
    @PrimaryKey
    @ColumnInfo(name = "tag")
    var tag: String = ""
) {
    // Constructor for creating with tag
    @Ignore
    constructor(tag: String, dummy: Boolean) : this(tag)
}
